File: agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/SerializedChatTranscript.kt
@file:Suppress("FunctionName", "ClassName", "unused", "EnumEntryName", "UnusedImport")
package com.sourcegraph.cody.agent.protocol_generated;

data class SerializedChatTranscript(
  val id: String,
  val chatTitle: String? = null,
  val interactions: List<SerializedChatInteraction>,
  val lastInteractionTimestamp: String,
  val enhancedContext: EnhancedContextParams? = null,
)

File: agent/bindings/csharp/SerializedChatTranscript.cs
using System.Text.Json.Serialization;

namespace Cody.Core.Agent.Protocol
{
  public class SerializedChatTranscript
  {
    [JsonProperty(PropertyName = "id")]
    public string Id { get; set; }
    [JsonProperty(PropertyName = "chatTitle")]
    public string ChatTitle { get; set; }
    [JsonProperty(PropertyName = "interactions")]
    public SerializedChatInteraction[] Interactions { get; set; }
    [JsonProperty(PropertyName = "lastInteractionTimestamp")]
    public string LastInteractionTimestamp { get; set; }
    [JsonProperty(PropertyName = "enhancedContext")]
    public EnhancedContextParams EnhancedContext { get; set; }
  }
}
